test/unit/filters/CookieConsentFilterSpec.scala

```diff
@@ -0,0 +1,85 @@
+import play.api.mvc.{Cookie, DefaultActionBuilder, EssentialAction}
+import play.api.test.Helpers.cookies
+import uk.gov.hmrc.trackingconsentfrontend.filters.CookieConsentFilter
+import unit.SpecBase
+
+import scala.concurrent.ExecutionContext
+
+class CookieConsentFilterSpec extends SpecBase {
+
+  val cookieConsentFilter = new CookieConsentFilter()(ExecutionContext.global)
+
+  implicit lazy val Action: DefaultActionBuilder = app.injector.instanceOf(classOf[DefaultActionBuilder])
+
+  "CookieConsentFilter" should {
+    "modify userConsent cookie" when {
+      "cookie is httpOnly and max age is defined" in {
+        val incorrectCookie         = Cookie(name = "userConsent", value = "cookieVal", maxAge = Some(100), httpOnly = true)
+        val expectedCookie          = Cookie(name = "userConsent", value = "cookieVal", maxAge = Some(100), httpOnly = false)
+        val request                 = FakeRequest().withCookies(incorrectCookie)
+        val action: EssentialAction = Action(_ => Ok("success"))
+
+        val result = cookieConsentFilter.apply(action)(request).run()
+
+        cookies(result).head mustBe expectedCookie
+      }
+
+      "cookie is httpOnly and max age is not defined" in {
+        val incorrectCookie         = Cookie(name = "userConsent", value = "cookieVal", httpOnly = true)
+        val expectedCookie          =
+          Cookie(name = "userConsent", value = "cookieVal", maxAge = Some(31556926), httpOnly = false)
+        val request                 = FakeRequest().withCookies(incorrectCookie)
+        val action: EssentialAction = Action(_ => Ok("success"))
+
+        val result = cookieConsentFilter.apply(action)(request).run()
+
+        cookies(result).head mustBe expectedCookie
+      }
+    }
+
+    "modify mdtpurr cookie" when {
+      "cookie is httpOnly and max age is defined" in {
+        val incorrectCookie         = Cookie(name = "mdtpurr", value = "cookieVal", maxAge = Some(100), httpOnly = true)
+        val expectedCookie          = Cookie(name = "mdtpurr", value = "cookieVal", maxAge = Some(100), httpOnly = false)
+        val request                 = FakeRequest().withCookies(incorrectCookie)
+        val action: EssentialAction = Action(_ => Ok("success"))
+
+        val result = cookieConsentFilter.apply(action)(request).run()
+
+        cookies(result).head mustBe expectedCookie
+      }
+
+      "cookie is httpOnly and max age is not defined" in {
+        val incorrectCookie         = Cookie(name = "mdtpurr", value = "cookieVal", httpOnly = true)
+        val expectedCookie          = Cookie(name = "mdtpurr", value = "cookieVal", maxAge = Some(31556926), httpOnly = false)
+        val request                 = FakeRequest().withCookies(incorrectCookie)
+        val action: EssentialAction = Action(_ => Ok("success"))
+
+        val result = cookieConsentFilter.apply(action)(request).run()
+
+        cookies(result).head mustBe expectedCookie
+      }
+    }
+  }
```

Comment:
See comment above - feels like we're missing a test that other cookies are unchanged.

Reply:
Added additional tests